### ISSUE TITLE
Weird behaviour when formatting file

### ISSUE THREAD

**Issue by mosheavni** [nvuillam/npm-groovy-lint#232]
Hi, thank you for the latest PR that allows formatting single files, however for some weird reason, it works sometimes, and other times it produces this:
https://asciinema.org/a/kcXVRlVVA1m0YC0r05a5C88Ck

![image](https://user-images.githubusercontent.com/17252601/184543530-27ce4480-0b3b-4979-93fc-88f037e37862.png)
```
❯ npm-groovy-lint --format null-ls_5JgVhJ.groovy
GroovyLint [░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░] Fixing {file}Unexpected error: ENOENT: no such file or directory, open '/Users/mavni/temp/groovy/null-ls_5JgVhJ.groovy'
Error: ENOENT: no such file or directory, open '/Users/mavni/temp/groovy/null-ls_5JgVhJ.groovy'
GroovyLint [████████████████████████████████████████] Fixing {file}


^C
```

At first I thought this is related to the filename being with dashes and underscores, but it happens even without them,
not sure how to debug further.
Thanks.

**Comment by nvuillam** [comment]
I have a look, thanks for reporting !

**Comment by nvuillam** [comment]
@mosheavni I succeeded to reproduce and fix the bug, please can you try again with v10.0.1 ?

**Comment by mosheavni** [comment]
> @mosheavni I succeeded to reproduce and fix the bug, please can you try again with v10.0.1 ?

It does work, however only if the file is NOT in `/tmp` directory.
Is this intended?
Can I format a file in `/tmp`?
Thanks.

**Comment by nvuillam** [comment]
What is the error when you try to format a file in /tmp ?

**Comment by nvuillam** [comment]
I have the same issue while trying to upgrade MegaLinter

investigating...

**Comment by nvuillam** [comment]
@mosheavni I published a new patch, v10.0.3 

As CodeNarc (the groovy linter behind npm-groovy-lint) can not take a list of files as input parameters, but only a base directory + Ant-like patterns, I had to hack to find the biggest common ancestor directory of the input list of files... it's still not optimal, and when then base directory contain too many files I think CodeNarc browses all of it anyway. 

I started a discussion (https://github.com/CodeNarc/CodeNarc/issues/714) to see how we could improve that

Meanwhile... please can you test again with v10.0.3 ? :)

**Comment by mosheavni** [comment]
it's working! thank you!!
I will however post another issue for wrong formatting on some places :)
but again, kudos for this project, it addresses a problem many experience without a real solution until now.

**Comment by nvuillam** [comment]
@mosheavni I'm glad your use case is solved :)

Thanks for the kudos ❤️ 
Please report any issue you see ... I created npm-groovy-lint when I used Groovy at work, but I don't do it anymore so there are probably lots of cases I've not met while making my tests :)

**Comment by mosheavni** [comment]
> @mosheavni I'm glad your use case is solved :)
> 
> Thanks for the kudos ❤️ Please report any issue you see ... I created npm-groovy-lint when I used Groovy at work, but I don't do it anymore so there are probably lots of cases I've not met while making my tests :)

Can you actually solve wrong formatting or it's related solely to CodeNarc?
Happy holidays BTW :)

**Comment by nvuillam** [comment]
@mosheavni CodeNarc detects, then npm-groovy-lint parses returned errors, then applies formatting & fixes
Parsing CodeNarc results also allow to identify exact range so errors can be highlighted with more precision in VsCode Groovy Lint

All CodeNarc rules results parsing & fixes can be seen here , in case someday you'd like to make a Pull Request :)
https://github.com/nvuillam/npm-groovy-lint/tree/master/lib/rules

Example of a rule here : https://github.com/nvuillam/npm-groovy-lint/blob/master/lib/rules/SpaceInsideParentheses.js
